metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestGroupsStep.java

```diff
@@ -0,0 +1,95 @@
+package com.linkedin.metadata.boot.steps;
+
+import com.datahub.util.RecordUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.identity.CorpGroupInfo;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.boot.BootstrapStep;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.models.AspectSpec;
+import com.linkedin.metadata.utils.EntityKeyUtils;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.GenericAspect;
+import com.linkedin.mxe.MetadataChangeProposal;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.ClassPathResource;
+
+import static com.linkedin.metadata.Constants.*;
+
+
+@Slf4j
+@RequiredArgsConstructor
+public class IngestGroupsStep implements BootstrapStep {
+  private final EntityService _entityService;
+
+  @Override
+  public String name() {
+    return "IngestGroupsStep";
+  }
+
+  @Override
+  public void execute() throws IOException, URISyntaxException {
+
+    final ObjectMapper mapper = new ObjectMapper();
+
+    // 0. Execute preflight check to see whether we need to ingest groups
+    log.info("Ingesting default groups...");
+
+    // 1. Read from the file into JSON.
+    final JsonNode groupsObj = mapper.readTree(new ClassPathResource("./boot/groups.json").getFile());
+
+    if (!groupsObj.isArray()) {
+      throw new RuntimeException(
+          String.format("Found malformed groups file, expected an Array but found %s", groupsObj.getNodeType()));
+    }
+
+    // 2. For each JSON object, cast into a CorpGroupInfo object.
+    for (final JsonNode groupObj : groupsObj) {
+      final Urn urn = Urn.createFromString(groupObj.get("urn").asText());
+
+      // If the info is not there, it means that the group was there before, but must now be removed
+      if (!groupObj.has("info")) {
+        _entityService.deleteUrn(urn);
```

Comment:
Whoa whoa whoa. This means the same groups are reingested eveery time DataHub starts. We don't want this. If a user decides to remove these default groups, they should stay gone.

Reply:
ah yeah that actually seems like reasonable behavior, will remove